varname/ignore.py

```diff
@@ -140,13 +133,16 @@ class IgnoreFunction(IgnoreElem):
     def __init__(self, ignore: FunctionType) -> None:
         super().__init__(ignore)
         if (
-                # but this is annoying when the whole set (decorated function +
-                # var = func()) inside the same function
-                #'<locals>' in ignore.__qualname__ or # without functools.wraps
+                '<locals>' in ignore.__qualname__ or # without functools.wraps
                 ignore.__name__ != ignore.__code__.co_name
         ):
-            warnings.warn('A decorated function may be used '
-                          'as an ignore element directly for varname.')
+            warnings.warn(
+                f'You asked varname to ignore function {ignore.__name__!r}, '
+                'which may be decorated. If it is not intended, you may need '
+                'to ignore all intemediated frames or with a tuple of '
+                'the function and the number of its decorators.',
```

Comment:
I don't think we need to teach people how to suppress the warnings in python here.